From 79970095815321c66a47bcb71a5d612e71957609 Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@cloudera.com>
Date: Tue, 11 Dec 2012 14:00:04 -0800
Subject: [PATCH 169/795] MR1: CLOUDERA-BUILD. Fix DataChecksum API usage to reflect HADOOP-8700 changes

Reason: Fix the build
Ref: CDH-9432
Author: Karthik Kambatla
(cherry picked from commit 5ccf61ab49ccd89813b3ecc1bac6504c7a9575d1)
(cherry picked from commit 2caaaaaf072aed1764ca5f087be04c6935236f44)
---
 .../org/apache/hadoop/mapred/IFileInputStream.java |    3 ++-
 .../apache/hadoop/mapred/IFileOutputStream.java    |    3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/mapred/org/apache/hadoop/mapred/IFileInputStream.java b/src/mapred/org/apache/hadoop/mapred/IFileInputStream.java
index 5495e4a..2febf25 100644
--- a/src/mapred/org/apache/hadoop/mapred/IFileInputStream.java
+++ b/src/mapred/org/apache/hadoop/mapred/IFileInputStream.java
@@ -83,7 +83,8 @@
   public IFileInputStream(InputStream in, long len, Configuration conf) {
     this.in = in;
     this.inFd = getFileDescriptorIfAvail(in);
-    sum = DataChecksum.newDataChecksum(DataChecksum.CHECKSUM_CRC32, 
+    sum = DataChecksum.newDataChecksum(
+        DataChecksum.Type.valueOf(DataChecksum.CHECKSUM_CRC32),
         Integer.MAX_VALUE);
     checksumSize = sum.getChecksumSize();
     length = len;
diff --git a/src/mapred/org/apache/hadoop/mapred/IFileOutputStream.java b/src/mapred/org/apache/hadoop/mapred/IFileOutputStream.java
index 84e022a..b1c9eb9 100644
--- a/src/mapred/org/apache/hadoop/mapred/IFileOutputStream.java
+++ b/src/mapred/org/apache/hadoop/mapred/IFileOutputStream.java
@@ -45,7 +45,8 @@
    */
   public IFileOutputStream(OutputStream out) {
     super(out);
-    sum = DataChecksum.newDataChecksum(DataChecksum.CHECKSUM_CRC32,
+    sum = DataChecksum.newDataChecksum(
+        DataChecksum.Type.valueOf(DataChecksum.CHECKSUM_CRC32),
         Integer.MAX_VALUE);
     barray = new byte[sum.getChecksumSize()];
   }
-- 
1.7.0.4

